File: App/app/src/main/java/com/cinema/ticket_booking/ui/staff/CheckInHistoryAdapter.java
package com.cinema.ticket_booking.ui.staff;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.cinema.ticket_booking.R;
import com.cinema.ticket_booking.data.model.response.CheckInHistoryItemResponse;

import java.util.ArrayList;
import java.util.List;

public class CheckInHistoryAdapter extends RecyclerView.Adapter<CheckInHistoryAdapter.ViewHolder> {

    private List<CheckInHistoryItemResponse> items = new ArrayList<>();

    public void submitList(List<CheckInHistoryItemResponse> list) {
        this.items = list != null ? list : new ArrayList<>();
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_check_in_history, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        CheckInHistoryItemResponse item = items.get(position);

        // Movie title
        holder.tvMovieTitle.setText(item.movieTitle != null ? item.movieTitle : "Không xác định");

        // Screen + Seats
        String screenSeats = "";
        if (item.screenName != null) screenSeats += item.screenName;
        if (item.seatsChecked != null && !item.seatsChecked.isEmpty()) {
            screenSeats += " • " + item.seatsChecked;
        }
        holder.tvScreenAndSeats.setText(screenSeats.isEmpty() ? "-" : screenSeats);

        // Customer name
        holder.tvCustomerName.setText(item.customerName != null ? item.customerName : "Khách vãng lai");

        // Time
        String timeDisplay = "--:--";
        if (item.scannedAt != null && item.scannedAt.length() >= 16) {
            timeDisplay = item.scannedAt.substring(11, 16);
        }
        holder.tvScannedAt.setText(timeDisplay);

        // Status badge
        if (item.success) {
            holder.tvStatusBadge.setText("✓ Thành công");
            holder.tvStatusBadge.setTextColor(0xFF22C55E);
            holder.tvStatusBadge.setBackgroundResource(R.drawable.bg_badge_success);
            holder.tvFailReason.setVisibility(View.GONE);
        } else {
            holder.tvStatusBadge.setText("✗ Thất bại");
            holder.tvStatusBadge.setTextColor(0xFFEF4444);
            holder.tvStatusBadge.setBackgroundResource(R.drawable.bg_badge_fail);
            if (item.failReason != null && !item.failReason.isEmpty()) {
                holder.tvFailReason.setVisibility(View.VISIBLE);
                holder.tvFailReason.setText("↳ " + item.failReason);
            } else {
                holder.tvFailReason.setVisibility(View.GONE);
            }
        }

        // Movie poster
        if (item.moviePosterUrl != null && !item.moviePosterUrl.isEmpty()) {
            Glide.with(holder.ivMoviePoster.getContext())
                    .load(item.moviePosterUrl)
                    .placeholder(R.color.surface)
                    .centerCrop()
                    .into(holder.ivMoviePoster);
        } else {
            holder.ivMoviePoster.setImageResource(R.color.surface);
        }
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        ImageView ivMoviePoster;
        TextView tvMovieTitle, tvScreenAndSeats, tvCustomerName, tvScannedAt,
                tvStatusBadge, tvFailReason;

        ViewHolder(@NonNull View itemView) {
            super(itemView);
            ivMoviePoster = itemView.findViewById(R.id.ivMoviePoster);
            tvMovieTitle = itemView.findViewById(R.id.tvMovieTitle);
            tvScreenAndSeats = itemView.findViewById(R.id.tvScreenAndSeats);
            tvCustomerName = itemView.findViewById(R.id.tvCustomerName);
            tvScannedAt = itemView.findViewById(R.id.tvScannedAt);
            tvStatusBadge = itemView.findViewById(R.id.tvStatusBadge);
            tvFailReason = itemView.findViewById(R.id.tvFailReason);
        }
    }
}

File: App/app/src/main/java/com/cinema/ticket_booking/ui/staff/CheckInHistoryAdapter.kt
package com.cinema.ticket_booking.ui.staff

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.cinema.ticket_booking.R
import com.cinema.ticket_booking.data.model.response.CheckInHistoryItemResponse
import java.util.ArrayList

class CheckInHistoryAdapter : RecyclerView.Adapter<CheckInHistoryAdapter.ViewHolder>() {

    private var items: List<CheckInHistoryItemResponse> = ArrayList()

    fun submitList(list: List<CheckInHistoryItemResponse>?) {
        this.items = list ?: ArrayList()
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_check_in_history, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = items[position]

        // Movie title
        holder.tvMovieTitle.text = item.movieTitle ?: "Không xác định"

        // Cinema name
        holder.tvCinemaName.text = item.cinemaName ?: "Hệ thống Nova"

        // Screen + Seats
        var screenSeats = ""
        if (item.screenName != null) screenSeats += item.screenName
        if (!item.seatsChecked.isNullOrEmpty()) {
            screenSeats += " • " + item.seatsChecked
        }
        holder.tvScreenAndSeats.text = if (screenSeats.isEmpty()) "-" else screenSeats

        // Booking code
        holder.tvBookingCode.text = if (item.bookingCode != null) "#" + item.bookingCode else ""

        // Customer name
        holder.tvCustomerName.text = item.customerName ?: "Khách vãng lai"

        // Time
        var timeDisplay = "--:--"
        val scannedAt = item.scannedAt
        if (scannedAt != null && scannedAt.length >= 16) {
            timeDisplay = scannedAt.substring(11, 16)
        }
        holder.tvScannedAt.text = timeDisplay

        // Status badge
        if (item.success) {
            holder.tvStatusBadge.text = "✓ Thành công"
            holder.tvStatusBadge.setTextColor(0xFF22C55E.toInt())
            holder.tvStatusBadge.setBackgroundResource(R.drawable.bg_badge_success)
            holder.tvFailReason.visibility = View.GONE
        } else {
            holder.tvStatusBadge.text = "✗ Thất bại"
            holder.tvStatusBadge.setTextColor(0xFFEF4444.toInt())
            holder.tvStatusBadge.setBackgroundResource(R.drawable.bg_badge_fail)
            val failReason = item.failReason
            if (!failReason.isNullOrEmpty()) {
                holder.tvFailReason.visibility = View.VISIBLE
                holder.tvFailReason.text = "↳ $failReason"
            } else {
                holder.tvFailReason.visibility = View.GONE
            }
        }

        // Movie poster
        val moviePosterUrl = item.moviePosterUrl
        if (!moviePosterUrl.isNullOrEmpty()) {
            Glide.with(holder.ivMoviePoster.context)
                .load(moviePosterUrl)
                .placeholder(R.color.surface)
                .centerCrop()
                .into(holder.ivMoviePoster)
        } else {
            holder.ivMoviePoster.setImageResource(R.color.surface)
        }
    }

    override fun getItemCount(): Int = items.size

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val ivMoviePoster: ImageView = itemView.findViewById(R.id.ivMoviePoster)
        val tvMovieTitle: TextView = itemView.findViewById(R.id.tvMovieTitle)
        val tvCinemaName: TextView = itemView.findViewById(R.id.tvCinemaName)
        val tvScreenAndSeats: TextView = itemView.findViewById(R.id.tvScreenAndSeats)
        val tvBookingCode: TextView = itemView.findViewById(R.id.tvBookingCode)
        val tvCustomerName: TextView = itemView.findViewById(R.id.tvCustomerName)
        val tvScannedAt: TextView = itemView.findViewById(R.id.tvScannedAt)
        val tvStatusBadge: TextView = itemView.findViewById(R.id.tvStatusBadge)
        val tvFailReason: TextView = itemView.findViewById(R.id.tvFailReason)
    }
}
